File: web-client/src/app/domain/entities/tournament-entity.ts
import {UserEntity} from './user-entity';
import {GameEntity} from './game-entity';

export class TournamentEntity {

  constructor(
    public readonly id: number,
    public readonly playersNumber: number,
    public readonly title: string,
    public readonly description: string,
    public readonly mode: string,
    public readonly admin: UserEntity,
    public readonly game: GameEntity
  ) {}
}

File: server-backend/src/main/kotlin/unito/taas/project/data/TournamentEntity.kt
package unito.taas.project.data

import javax.persistence.*
import javax.validation.constraints.NotBlank

@Entity
data class TournamentEntity(
        @Id @GeneratedValue val id: Long,
        @NotBlank val playersNumber: Int,
        @NotBlank val title: String,
        @NotBlank val tournamentDescription: String,
        @NotBlank val tournamentMode: String = "",
        @NotBlank @JoinColumn(name = "admin") val admin: String,
        @NotBlank @ManyToOne @JoinColumn(name = "game") val game: GameEntity
)